source/common/network/win32_socket_handle_impl.cc

```diff
@@ -18,16 +18,24 @@ namespace Network {
 
 Api::IoCallUint64Result Win32SocketHandleImpl::readv(uint64_t max_length, Buffer::RawSlice* slices,
                                                      uint64_t num_slice) {
-  auto result = IoSocketHandleImpl::readv(max_length, slices, num_slice);
-  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
-  return result;
+  if (peek_buffer_->length() == 0) {
+    auto result = IoSocketHandleImpl::readv(max_length, slices, num_slice);
+    reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
+    return result;
+  }
+
+  return readvFromPeekBuffer(max_length, slices, num_slice);
```

Comment:
we need to re-enable the event for this also

Reply:
My understanding is that the caller should be responsible to read in a loop until it `EWOULDBLOCK` is returned and if it doesn't the caller is at fault. 

Following that logic at some point `peek_buffer_->length() == 0` would be hit and then on the last call we re-enable

Reply:
this is a recurring comment so I only reply here and we can handle all the cases when we reach a concensus.

Reply:
ah, you are right, got it.